exasol_script_languages_container_ci/lib/ci.py

```diff
@@ -36,20 +60,28 @@ def ci(ctx: click.Context,
     push_to_public_cache = bool(IS_MASTER.match(branch_name))
 
     flavor_path = (f"flavors/{flavor}",)
-    ci_build(ctx, flavor_path=flavor_path, rebuild=rebuild, build_docker_repository=docker_build_repository,
-          commit_sha=commit_sha,
-          docker_user=docker_user, docker_password=docker_password)
-
-    ci_test(ctx, flavor_path=flavor_path)
-    ci_security_scan(ctx, flavor_path=flavor_path)
-    ci_push(ctx, flavor_path=flavor_path,
-         target_docker_repository=docker_build_repository, target_docker_tag_prefix=commit_sha,
-         docker_user=docker_user, docker_password=docker_password)
-    ci_push(ctx, flavor_path=flavor_path,
-         target_docker_repository=docker_build_repository, target_docker_tag_prefix="",
-         docker_user=docker_user, docker_password=docker_password)
-
-    if push_to_public_cache:
+
+    need_to_run = rebuild or check_if_need_to_build(config_file, flavor)
```

Comment:
what is with release r is this handle by master

Reply:
The relase build will be implemented in another ticket.